eng/pipelines/templates/stages/vscode-build-and-test.yml

```diff
@@ -43,3 +43,68 @@ stages:
               OSVmImage: ${{ build.value.OSVmImage }}
               OS: ${{ build.value.OS }}
               Variables: ${{ build.value.Variables }} 
+      
+      - job: GenerateReleaseArtifacts
+        pool:
+          name: $(LINUXPOOL)
+          image: $(LINUXVMIMAGE)
+          os: linux
+
+        steps:
+          - checkout: self
+
+          - ${{ if in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI') }}:
+            - pwsh: |
+                $packageJson = Get-Content -Raw ext/vscode/package.json
+                $package = ConvertFrom-Json $packageJson
+                $package.version = "$($package.version)-beta.$(Build.BuildId)"
+                $outputContent = ConvertTo-Json $package -Depth 100
+                Set-Content -Path ext/vscode/package.json -Value $outputContent
+              displayName: Set Dev Version in package.json
+
+          - task: PowerShell@2
+            inputs:
+              pwsh: true
+              targetType: filePath
+              filePath: eng/scripts/Set-VsixVersionVariable.ps1
+            displayName: Set VSIX_VERSION
+
+          # Changelog
+          - pwsh: New-Item -ItemType Directory -Path changelog/
+            displayName: Create changelog directory
+
+          - task: PowerShell@2
+            inputs:
+              targetType: filePath
+              filePath: eng/scripts/New-ReleaseNotesFile.ps1
+              arguments: >-
+                -ChangeLogPath ext/vscode/CHANGELOG.md
+                -Version $(VSIX_VERSION)
+                -OutputPath changelog/CHANGELOG.md
+            displayName: Set release changelog content
+
+          # Release metadata
+          - pwsh: |
+              $packageJson = Get-Content -Raw ext/vscode/package.json
+              $package = ConvertFrom-Json $packageJson
+
+              $releaseMetadata = @{ 
+                vsixVersion = "$(VSIX_VERSION)";
+                packageVersion = $package.version;
+                ghReleaseTag = 'azure-dev-vscode_$(VSIX_VERSION)';
+              }
+              New-Item -ItemType Directory -Path release-metadata
+              $releaseMetadata | ConvertTo-Json -Depth 100 | Set-Content -Path release-metadata/metadata.json
+            displayName: Set Release Metadata
+
+        templateContext:
+          outputs: 
+            - output: pipelineArtifact
+              path: release-metadata
+              condition: succeeded()
+              artifact: release-metadata
+              displayName: Upload release metadata
+
+            - output: pipelineArtifact
```

Comment:
Why is the changelog going as an independent artifact?

Reply:
Changelog is used to create the GitHub release. The GitHub release also includes signed artifacts. Rather than break up the creation of the GitHub Release and the uploading of artifacts, I just made the release creation and artifact upload a deployment job.